test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs

```diff
@@ -7,15 +7,66 @@
 using System.Collections.Immutable;
 using System.IO;
 using System.Linq;
+using System.Threading.Tasks;
+using ILLink.CodeFix;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Text;
 using Xunit;
 
 namespace ILLink.RoslynAnalyzer.Tests
 {
 	public class TestCaseUtils
 	{
+		private const string _requiresAssemblyFilesAttributeDefinition = @"
+#nullable enable
+namespace System.Diagnostics.CodeAnalysis
+{
+	[AttributeUsage(AttributeTargets.Constructor | AttributeTargets.Event | AttributeTargets.Method | AttributeTargets.Property, Inherited = false, AllowMultiple = false)]
+	public sealed class RequiresAssemblyFilesAttribute : Attribute
+	{
+		public RequiresAssemblyFilesAttribute () { }
+		public string? Message { get; set; }
+		public string? Url { get; set; }
+	}
+}";
+		private const string _requiresUnreferencedCodeAttributeDefinition = @"
```

Comment:
Why are we defining this and `UnconditionalSuppress` ourselves?

Reply:
We define our own analyzer verifier so we pass the metadata references into our compilation https://github.com/mono/linker/blob/main/test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier%601.cs#L48 in the case of the codefix we rely on the code in Roslyn which has references to Microsoft.NETCore.App.Ref.3.1.0 so we needed to include our own definition

Reply:
Let's just update the references to 5.0 then.

Reply:
I was experimenting with the SolutionTransforms and I just manage to do the same as we do on analyzers tests, get the metadata from the SPC in the linker which is 6.0